lib/metadata_json_lint.rb

```diff
@@ -136,10 +142,34 @@ def validate_requirements!(requirements)
       if requirement['name'] == 'pe'
         warn :requirements, "The 'pe' requirement is no longer supported by the Forge."
       end
+
+      begin
+        puppet_req = VersionRequirement.new(requirement.fetch('version_requirement', ''))
+      rescue ArgumentError => e
+        # Raised when the version_requirement provided could not be parsed
+        error :dependencies, "Invalid 'version_requirement' field in metadata.json: #{e}"
+      end
+
+      validate_puppet_ver!(puppet_req)
     end
   end
   module_function :validate_requirements!
 
+  def validate_puppet_ver!(requirement)
+    if options[:strict_puppet_version] && requirement.open_ended?
+      warn(:requirements, "Puppet has an open ended version requirement #{requirement.ver_range}")
+    end
+
+    if options[:strict_puppet_version] && requirement.puppet_eol?
+      warn(:requirements, "#{requirement.min} is no longer supported. Minimum supported version is 4.10.0")
```

Comment:
I really don't like hardcoded versions. It's just waiting to be outdated. It should at least be a constant somewhere.